src/apprt/gtk/wayland.zig

```diff
@@ -0,0 +1,89 @@
+const std = @import("std");
+const c = @import("c.zig").c;
+const wayland = @import("wayland");
+const wl = wayland.client.wl;
+const org = wayland.client.org;
+
+const log = std.log.scoped(.gtk_wayland);
+
+const Wayland = @This();
+
+surface: *wl.Surface,
+
+blur_manager: ?*org.KdeKwinBlurManager = null,
+blur_token: ?*org.KdeKwinBlur = null,
+
+pub fn init(window: *c.GtkWindow) !?Wayland {
+    const surface = c.gtk_native_get_surface(@ptrCast(window)) orelse return null;
+    const display = c.gdk_surface_get_display(surface) orelse return null;
+
+    // Check if we're actually on Wayland
+    if (c.g_type_check_instance_is_a(
+        @ptrCast(@alignCast(surface)),
+        c.gdk_wayland_surface_get_type(),
+    ) == 0)
+        return null;
+
+    const wl_surface: *wl.Surface = @ptrCast(c.gdk_wayland_surface_get_wl_surface(surface) orelse return null);
+    const wl_display: *wl.Display = @ptrCast(c.gdk_wayland_display_get_wl_display(display) orelse return null);
+
+    const registry = try wl_display.getRegistry();
+
+    var self = Wayland{ .surface = wl_surface };
+
+    registry.setListener(*Wayland, registryListener, &self);
```

Comment:
I'm not familiar with Wayland protocols [yet] but is this safe to do because the callback happens just below this? Otherwise, `&self` is a pointer to the stack.

Reply:
this is fine for now because the roundtrip blocks on the events being processed https://wayland.freedesktop.org/docs/html/apb.html#Client-classwl__display_1ab60f38c2f80980ac84f347e932793390 but in the future we might want a stable pointer here, for example in my apprt it is very useful that the surface is a stable pointer on init so you can do things like this https://github.com/gabydd/ghostty-wayland/blob/ab63027d485eeee31124bb1cd52a0e877ef5ea13/src/wayland.zig#L772-L773 without much song and dance, whereas things at the app level are more complicated though that is mostly because i have to get the Surface from the id of the wayland surface which shouldnt be a problem here

Reply:
This is fine if we do a roundtrip (which is blocking). I had run into bugs where a listener is accidentally using stale pointers, so we should definitely be cautious with that (maybe with a comment explaining why this is fine)

Reply:
If a future use case is going to require a stable pointer anyways, we should go ahead and do it now. Bolting on a pointer stability later kind of sucks. We've done it a lot in the codebase.